### PR TITLE
Update golang to 1.22.

### PR DESCRIPTION
**Pull request by jkh52**
Update golang to 1.22.

(Defer konnectivity-client to separate PR).

### CODE REVIEW

**Comment by jkh52** [comment]
This PR will be unblocked by https://github.com/kubernetes/test-infra/pull/32540

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ on:
 
 env:
   GO_TOOLCHAIN: "golang"
-  GO_VERSION: "1.21.6"
+  GO_VERSION: "1.22.2"
```
Comment (liangyuanpeng) [inline .github/workflows/e2e.yaml:15]: Woud you want to delete this param and let's setup-go version with go.mod?

```yaml
      - uses: actions/setup-go@v5
        with:
          go-version-file: go.mod
```

Comment (jkh52) [reply .github/workflows/e2e.yaml:15]: I agree, that's nicer. I'll wait for the test-infra PR to merge, then update here.

**Comment by cheftako** [comment]
I've lgtm'd the test-infra PR.

File: .github/workflows/e2e.yaml (hunk 1 above)
Comment (jkh52) [reply .github/workflows/e2e.yaml]: Done.

**Pull request merged**